### PR TITLE
4693 enable partitioning in bulk import

### PR DESCRIPTION
**Pull request by StevenXLi**
- added partitionid to jobinstance via PartitionablePartitionId
- changed functionality in cosumefilestep to support mutitenancy

### CODE REVIEW

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/imprt/BulkImportJobParameters.java [hunk 1]
```diff
@@ -51,6 +52,10 @@ public class BulkImportJobParameters implements IModelJson {
 	@Nullable
 	private Integer myMaxBatchResourceCount;
 
+	@JsonProperty(value = "partitionId", required = false)
```
Comment (michaelabuckley) [inline hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/imprt/BulkImportJobParameters.java:55]: I don't understand why we have the RequestPartitionId in both the JobInstance, and here.  JobInstance is available in StepExecutionDetails. Can we just do this in one place?

**Review by michaelabuckley — CHANGES_REQUESTED** [review]
This looks very clean, but we don't need two copies of the RequestPartitionId.  How about adding it to JobInstanceStartRequest and JobInstance, and the parameters don't need it.

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/model/JobInstance.java [hunk 2]
```diff
@@ -102,6 +103,9 @@ public class JobInstance extends JobInstanceStartRequest implements IModelJson,
 	@JsonProperty(value = "report", access = JsonProperty.Access.READ_WRITE)
 	private String myReport;
 
+	@JsonProperty(value = "partitionId", access = JsonProperty.Access.READ_WRITE)
+	private PartitionablePartitionId myPartitionId;
```
Comment (michaelabuckley) [inline hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/model/JobInstance.java:107]: I don't see this used at all.

Comment (StevenXLi) [reply hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/model/JobInstance.java:107]: This is being used in `importPollStatus` in `BulkDataImportProvider`

File: hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/imprt/BulkImportJobParameters.java (hunk 1 above)
Comment (StevenXLi) [reply hapi-fhir-storage-batch2-jobs/src/main/java/ca/uhn/fhir/batch2/jobs/imprt/BulkImportJobParameters.java:55]: `JobInstance` is available in `StepExecutionDetails`, however it's not available in `importByManifest` in `BulkDataImportProvider` where it needs to be set in order to be used in the `ConsumeFilesStep`

**Review by michaelabuckley — APPROVED** [review]
Looks good.

**Pull request merged**